### PR TITLE
[task-6.1] Trend calculation service for digest emails

### PR DESCRIPTION
**Pull request by onlineeric**
## Summary
- Implement trend calculation service for price digest emails
- Calculate 7/30/90/180 day average prices and percentage changes
- Add test script for verification
- Clean up template backup directory

### New Files
- `apps/worker/src/services/trendCalculator.ts` - Service to calculate price trends
  - `calculateTrendsForAllProducts()` - Calculate trends for all active products
  - `calculateTrendsForProduct()` - Calculate trends for single product
  - Helper functions for averages and percentage changes
- `apps/worker/src/test-trends.ts` - Test script to verify trend calculations

### Updated Files
- `apps/worker/package.json` - Add date-fns dependency
- `pnpm-lock.yaml` - Update lockfile

### Removed Files
- `apps/web_backup/` - Delete template backup directory

## Features
- **Multi-period averages**: 7, 30, 90, 180 day price averages
- **Percentage changes**: Compare current price vs last check and historical averages
- **Edge case handling**: Gracefully handle products with no/insufficient data
- **Null safety**: Proper null checks to prevent NaN in calculations

## Test Plan
- [ ] Run test script: `cd apps/worker && npx tsx src/test-trends.ts`
- [ ] Verify all active products are listed
- [ ] Check price display formatting (cents to dollars)
- [ ] Confirm percentage changes calculate correctly
- [ ] Test edge cases (no data, single price, failed products)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/worker/src/services/trendCalculator.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+  // Get all active products
+  const allProducts = await db
+    .select()
+    .from(products)
+    .where(eq(products.active, true));
+
+  console.log(`[Trend Calculator] Found ${allProducts.length} active products`);
+
+  // Calculate trends for each product
+  const trendsData = await Promise.all(
```
Comment (onlineeric) [inline apps/worker/src/services/trendCalculator.ts]: **N+1 Query Problem - Severe Performance Issue**
**Severity:** Critical

**Problem:** The trend calculation performs 6 separate database queries per product (1 latest + 1 previous + 4 time periods), resulting in an N+1 query antipattern. For 50 products, this executes ~300 queries sequentially within Promise.all, causing severe database load and slow response times.

**Evidence:**
Lines 69-83: Two separate queries for latest/previous prices
Lines 97-118: Loop executing 4 more queries per product (one per time period)

With 50 products, this becomes:
- 50 queries for latest prices
- 50 queries for previous prices  
- 200 queries for time period averages (50 products × 4 periods)
= 300 total queries

**Fix hint:** Batch all queries using SQL joins or fetch all price records for each product in a single query, then calculate averages in-memory. Alternatively, use database aggregation functions (AVG, GROUP BY) to calculate all periods in one query per product.

**Comment by onlineeric** [comment]
**Massive Code Duplication Between Two Functions**
**Severity:** High

**Problem:** The functions `calculateTrendsForAllProducts` and `calculateTrendsForProduct` contain 100+ lines of identical logic (lines 173-250 duplicate lines 69-145). This violates DRY principle and creates maintenance risk - any bug fix or logic change must be applied twice.

**Evidence:**
- `calculateTrendsForAllProducts` (lines 69-145): Query logic for latest/previous/periods + trend calculation
- `calculateTrendsForProduct` (lines 173-250): Exact same logic duplicated

**Fix hint:** Extract the core trend calculation logic into a private helper function that both public functions call. Only the initial product fetching logic differs (all products vs single product by ID).
